advanced_functionality/pytorch_extending_our_containers/utils/setup.sh

```diff
@@ -1,5 +1,13 @@
 #!/bin/bash
 
+sudo -n true
+if [ $? -eq 0 ]; then
+  echo "The user has root access."
+else
+  echo "The user does not have root access. Everything required to run the notebook is already installed and setup. We are good to go!"
```

Comment:
Where comes from the assumption that if user doesn't have root access then everything is already installed and setup?

Reply:
+1

This check should also be moved below the `Do we have GPU support` check. We don't need to fail in the absence of root access if the dependencies are already present.

Reply:
@nadiaya, 
Previously we had a long term bug on the services side around more than a year which resulted in race condition where sometime required setup was already installed, sometimes not. Now we have a fix for it going along with removing of root permissions. It's a prerequisite to remove root permissions as some of the things which we are doing in the script are impossible to do without having sudo permissions. That's from where assumption is coming up. 

Secondly as the script is idempotent if the root user runs this script multiple times it has no effect but for root free user this script cannot be run. We just cannot remove setup.sh script for one of the reason is sample notebooks executed outside of SageMaker won't be able to execute the notebook which requires setup.sh file.

@jaipreet-s I don't think we are failing in absence of root access, its an `exit 0` code.